### PR TITLE
Update HISTORY.md to 5.12.0

### PR DESCRIPTION
**Pull request by miasantreble**

### CODE REVIEW

**Comment by gfosco** [comment]
I thought the 5.12 changes should stay in the .fb branch until landed?

File: HISTORY.md [hunk 1]
```diff
@@ -1,20 +1,46 @@
...
+
+### New Features
+* Add a new histogram stat called rocksdb.db.flush.micros for memtable flush.
+* Add "--use_txn" option to use transactional API in db_stress.
+* Disable onboard cache for compaction output in Windows platform.
+* Improve the performance of iterators doing long range scans by using readahead.
+
+### Bug Fixes
+* Fix a stack-use-after-scope bug in ForwardIterator.
+* Fix builds on platforms including Linux, Windows, and PowerPC.
```
Comment (ajkr) [inline HISTORY.md]: i'm confused, where are these lines coming from?

File: HISTORY.md [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
 * Creating checkpoint with empty directory now returns `Status::InvalidArgument`; previously, it returned `Status::IOError`.
-* RocksDBOptionsParser::Parse()'s `ignore_unknown_options` argument will only be effective if the option file shows it is generated using a higher version of RocksDB than the current version.
+* Adds a BlockBasedTableOption to turn off index block compression.
+* Close() method now returns a status when closing a db.
 
 ### New Features
 * Improve the performance of iterators doing long range scans by using readahead.
 * Add new function `DeleteFilesInRanges()` to delete files in multiple ranges at once for better performance.
+* FreeBSD build support for RocksDB and RocksJava.
+* Improved performance of long range scans with readahead.
```
Comment (ajkr) [inline HISTORY.md]: this is mentioned 3 lines above

File: HISTORY.md [hunk 3]
```diff
@@ -1,20 +1,46 @@
...
+
+## 5.11.0 (01/08/2018)
+### Public API Change
+* Add `autoTune` and `getBytesPerSecond()` to RocksJava RateLimiter
+
+### New Features
+* Add a new histogram stat called rocksdb.db.flush.micros for memtable flush.
+* Add "--use_txn" option to use transactional API in db_stress.
+* Disable onboard cache for compaction output in Windows platform.
+* Improve the performance of iterators doing long range scans by using readahead.
```
Comment (ajkr) [inline HISTORY.md]: this is also mentioned twice above

File: HISTORY.md (hunk 1 above)
Comment (miasantreble) [reply HISTORY.md]: from 5.11 release branch

File: HISTORY.md (hunk 2 above)
Comment (miasantreble) [reply HISTORY.md]: these are copied from 5.12 release branch. @gfosco can clean them up in both master and 5.12.fb branches.

**Pull request closed without merging**